docs/docs/ref-03-component-specs.md

```diff
@@ -37,6 +37,10 @@ object getInitialState()
 
 Invoked once before the component is mounted. The return value will be used as the initial value of `this.state`.
 
+> Note:
+>
+> This method is not available on ES6 `class` components that extend `React.Component`. It may be removed entirely in a future version of React.
```

Comment:
> You can do basic setting based on props with initializers.

Yup definitely. These are the docs pointing away from `getInitialProps` though, so I'd suggest the best analog: the constructor. I can imagine more questions if this instead pointed to property initializers asking how to do more complicated work for initial state than expressions will allow.